File: eCar/UI/ecar_admin/lib/models/Review/review.dart
import 'package:ecar_admin/models/Client/client.dart';
import 'package:ecar_admin/models/Driver/driver.dart';
import 'package:ecar_admin/models/Route/route.dart';
import 'package:json_annotation/json_annotation.dart';

part 'review.g.dart';

@JsonSerializable()
class Review {
  int? id;
  int? value;
  String? description;
  int? reviewsId;
  int? reviewedId;
  int? routeId;
  Driver? reviewed;
  Client? reviews;
  Route? route;

  Review(
      {this.id,
      this.value,
      this.description,
      this.reviewsId,
      this.reviewedId,
      this.routeId,
      this.reviewed,
      this.reviews,
      this.route});

  factory Review.fromJson(Map<String, dynamic> json) => _$ReviewFromJson(json);

  Map<String, dynamic> toJson() => _$ReviewToJson(this);
}

File: eCar/eCar.Model/Model/Review.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eCar.Model.Model
{
    public class Review
    {
        public int Id { get; set; }

        public int? Value { get; set; }

        public string? Description { get; set; }

        public int ReviewsId { get; set; }

        public int ReviewedId { get; set; }

        public int RouteId { get; set; }

        public virtual Driver Reviewed { get; set; } = null!;

        public virtual Client Reviews { get; set; } = null!;

        public virtual Route Route { get; set; } = null!;
    }
}
